server/src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -675,20 +727,17 @@ public Consumer createConsumerFromEntity(Consumer consumer, Principal principal,
             }
         }
 
-        if (consumer.getEntitlementCount() != 0) {
+        if (consumer.getEntitlementCount() != null || consumer.getEntitlementCount() != 0) {
```

Comment:
These type of checks should be unnecessary now -- they won't be leaving the dto.